devito/symbolics/queries.py

```diff
@@ -23,7 +23,7 @@
 
 
 def q_scalar(expr):
-    return expr.is_Number or expr.is_Symbol
+    return getattr(expr, 'is_Scalar', False)
```

Comment:
One should use `getattr` only when the attribute name is unknown , or perhaps in list comprehension (extreme case).

Otherwise, the pythonic way is

```
try:
    return expr.is_Scalar
except AttributeError:
    return False
```